backend/src/tasks/shodan.ts

```diff
@@ -0,0 +1,126 @@
+import { Domain, Service, Vulnerability } from '../models';
+import { plainToClass } from 'class-transformer';
+import getIps from './helpers/getIps';
+import { CommandOptions } from './ecs-client';
+import saveServicesToDb from './helpers/saveServicesToDb';
+import { chunk } from 'lodash';
+import axios from 'axios';
+import { IpToDomainsMap, sanitizeStringField } from './censysIpv4';
+import saveVulnerabilitiesToDb from './helpers/saveVulnerabilitiesToDb';
+
+// Shodan allows searching up to 100 IPs at once
+const CHUNK_SIZE = 100;
+
+interface ShodanResponse {
+  ip_str: string;
+  country_name: string;
+  domains: string[];
+  hostnames: string[];
+  org: string;
+  ports: number[];
+  os: string;
+  city: string;
+  longitude: number;
+  latitude: number;
+  data: {
+    port: number;
+    product: string;
+    data: string;
+    cpe: string[];
+    version: string;
+    vulns: {
+      [title: string]: {
+        verified: boolean;
+        references: string[];
+        cvss: number;
+        summary: string;
+      };
+    };
+  }[];
+}
+
+const sleep = (milliseconds) => {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+};
+
+export const handler = async (commandOptions: CommandOptions) => {
+  const { organizationId, organizationName } = commandOptions;
+
+  console.log('Running shodan on organization', organizationName);
+
+  const domainsWithIPs = await getIps(organizationId);
+
+  const ipToDomainsMap: IpToDomainsMap = domainsWithIPs.reduce<IpToDomainsMap>(
+    (map: IpToDomainsMap, domain: Domain) => {
+      if (!map[domain.ip]) {
+        map[domain.ip] = [];
+      }
+      map[domain.ip].push(domain);
+      return map;
+    },
+    {}
+  );
+
+  const chunks = chunk(domainsWithIPs, CHUNK_SIZE);
+
+  for (const domainChunk of chunks) {
+    let { data } = await axios.get<ShodanResponse[]>(
```

Comment:
Let's log progress on each iteration of this for loop (maybe just log the first domain) so we won't just see this scan stall on the logs.